### DISCUSSION TITLE
[Feature Request] Add ability to expand / contract all for structures

### DISCUSSION THREAD

**Discussion by mattcdavis1** [craftcms/cms#8576] [Ideas]
When you have a large structures data set and everything is expanded you have to manually keep scrolling and contract every entry.  It would be nice to be able to just click a button and have the whole tree contract.

**Comment by gioppe** [comment]
It would be super useful also for Commerce variants.

**Comment by narration-sd** [comment]
This is coming up often enough now, as persons are making increasingly complex sites with Craft -- a good thing! And as Vue etc. come into play, the need will advance, as it already shows with page items such as carousels on Twig templates -- you want to stop the moving, and concentrate on a slide.

I've sketched in #3116 what feels a quite nice and 'standard practice' solution. I think Brandon was attracted once I added some pictures, so the intent was more clear, so you may want to have a look.

What this is, is the feature you will find on every audio mixing board, from network television to software for your laptop: a solo button, one Solo button for each channel in the mix. 

This has two cases: push Solo, and you are listening to just the one track. Puah again, and it releases -- now you are listening to the mix of tracks just as it was set up before. 

It's probably pretty clear how this would map to Craft Matrix elements and other structures. Click to see just the one slide in the Carousel; click again once you're through editing it to see again all the enabled slides moving. 

The key is that Solo doesn't change what's enabled, etc...

As all things of this sort, there are going to be 'interesting' cases, but here I have the consistent confidence, as we all do. Might it be worthwhile to start producing this with Matrix, probably one of the most useful?

**Comment by McCrisp** [comment]
The ability to collapse the entire structure tree or a config to have a certain section collapsed by default would absolutely be useful! 

I'm working on a project right now with Shows and performances where shows are the main item, with all the information, and the performances are children of that item.

Performance wise its good, but trying to find shows and browse the structure is a pain, having the option to collapse everything here would be most welcome!

I know I can inject a JS-file and add the collapsed class that way, but it feels dirty.
